Sources/BraintreeCore/ConfigurationLoader.swift

```diff
@@ -39,26 +41,34 @@ class ConfigurationLoader {
             completion(cachedConfig, nil)
             return
         }
+        
+        pendingCompletions.append(completion)
+        
+        // If this is the 1st `v1/config` GET attempt, proceed with firing the network request.
+        // Otherwise, there is already a pending network request.
+        if pendingCompletions.count == 1 {
+            http.get(configPath, parameters: BTConfigurationRequest()) { [weak self] body, response, error in
+                guard let self else {
+                    self?.pendingCompletions.forEach { $0(nil, BTAPIClientError.deallocated) }
+                    self?.pendingCompletions.removeAll()
+                    return
+                }
 
-        http.get(configPath, parameters: BTConfigurationRequest()) { [weak self] body, response, error in
-            guard let self else {
-                completion(nil, BTAPIClientError.deallocated)
-                return
-            }
-
-            if let error {
-                completion(nil, error)
-                return
-            } else if response?.statusCode != 200 || body == nil {
-                completion(nil, BTAPIClientError.configurationUnavailable)
-                return
-            } else {
-                let configuration = BTConfiguration(json: body)
+                if let error {
+                    completion(nil, error)
```

Comment:
I meant don't we need to handle the pendingCompletions here?

Reply:
We definitely should! Updated